### PR TITLE
server: address golangci var-naming issues

### PR DESCRIPTION
**Pull request by ivanvc**
This PR addresses minor var-naming violations from the server directory.

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: server/mock/mockstore/store_recorder.go [hunk 1]
```diff
@@ -119,7 +119,11 @@ func (s *storeRecorder) Clone() v2store.Store {
 	return s
 }
 
+//revive:disable:var-naming
 func (s *storeRecorder) JsonStats() []byte { return nil }
+
+//revive:enable:var-naming
```
Comment (ivanvc) [inline server/mock/mockstore/store_recorder.go:125]: Added this ignores, as it comes from v2's `(*recorder) JsonStats().`

File: server/storage/backend/verify.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ import (
 )
 
 const (
-	ENV_VERIFY_VALUE_LOCK verify.VerificationType = "lock"
+	EnvVerifyValueLock verify.VerificationType = "lock"
```
Comment (ivanvc) [inline server/storage/backend/verify.go:29]: Even though this is exported, I just renamed it as it is used only by the `backend_tests` package, which is in the same directory but in a different package.

**Review by ahrtr — APPROVED** [review]
lgtm

Thanks

**Review by jmhbnz — APPROVED** [review]
Nice work @ivanvc, another one tidied up!

**Pull request merged**